### PR TITLE
fix: added errors to debugger

### LINKED ISSUE #7615: [Bug] Error message is displayed on debugger but error is not present
## Description
Error is displayed for linting but error message on debugger does not navigate to errors tab as is generally the case and no error is logged in the debugger.

### Steps to reproduce the behaviour:
1. Go to JS editor and type in any word to cause a lint error
2. Observe a parsing error displayed on the debugger - usual 'Debug' button is missing
3. Navigate to Errors tab and Logs and observe that no errors are logged

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Deploy preview on test.appsmith.com (JS editor)

[LOOM DEMO](https://www.loom.com/share/2c8081449e83406bad71b0257bd035e3)

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description

Errors were not showing up in debugger as we were filtering linting errors

Fixes #7615 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: show-errors-7615 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.81 **(-0.01)** | 36.81 **(-0.01)** | 33.89 **(0)** | 55.41 **(0)**
 :green_circle: | app/client/src/sagas/PostEvaluationSagas.ts | 17.61 **(0.47)** | 0 **(0)** | 0 **(0)** | 21.55 **(0.5)**
 :red_circle: | app/client/src/workers/evaluationUtils.ts | 73.31 **(-0.53)** | 58.87 **(0)** | 74.42 **(0)** | 71.31 **(-0.6)**</details>

### CODE REVIEW

File: app/client/src/workers/evaluationUtils.ts [hunk 1]
```diff
@@ -541,45 +541,55 @@ export const parseJSCollection = (
...
+      return {
+        evalTree,
+        result: {
+          actions: actions,
+          variables: variables,
+        },
+      };
+    } catch (e) {
+      return {
+        evalTree,
```
Comment (hetunandu) [inline app/client/src/workers/evaluationUtils.ts:590]: Why are we not returning result here? Nor errors

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/workers/evaluationUtils.ts (hunk 1 above)
Comment (ApekshaBhosale) [reply app/client/src/workers/evaluationUtils.ts:590]: because result is undefined and errors are set in evalTree alreay @hetunandu

Comment (hetunandu) [reply app/client/src/workers/evaluationUtils.ts:590]: This check is not handled in functions that are calling this

File: app/client/src/sagas/PostEvaluationSagas.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ const getDebuggerErrors = (state: AppState) => state.ui.debugger.errors;
  * W117: `x` is undefined
  */
 const errorCodesToIgnoreInDebugger = ["W117"];
+const errorCodesForJSEditorInDebugger = ["E041"];
```
[the remark this answers is not in the document]
Comment (ApekshaBhosale) [reply app/client/src/sagas/PostEvaluationSagas.ts]: this is jsHint throws something like "90% was parsed" which is not very useful for users

File: app/client/src/workers/evaluationUtils.ts (hunk 1 above)
Comment (ApekshaBhosale) [reply app/client/src/workers/evaluationUtils.ts:590]: what do you mean? we were already sending only evalTree in case of errors.

Comment (hetunandu) [reply app/client/src/workers/evaluationUtils.ts:590]: okay got it

**Review by hetunandu — APPROVED** [review]

**Pull request merged**